common/src/test/java/io/netty/util/NetUtilTest.java

```diff
@@ -87,7 +92,11 @@
             "[2001:0000:4136:e378:8000:63bf:3fff:fdd2]", "200100004136e378800063bf3ffffdd2",
             "0:1:2:3:4:5:6:789a",                        "0000000100020003000400050006789a",
             "0:1:2:3::f",                                "0000000100020003000000000000000f",
-            "0:0:0:0:0:0:10.0.0.1",                      "0000000000000000000000000a000001",
+            "0:0:0:0:0:0:10.0.0.1",                      "00000000000000000000ffff0a000001",
+            "0:0:0:0:0::10.0.0.1",                       "00000000000000000000ffff0a000001",
+            "::0:0:0:0:0:10.0.0.1",                      "00000000000000000000ffff0a000001",
+            "0:0:0:0:0:0::10.0.0.1",                     "00000000000000000000ffff0a000001",
```

Comment:
Yes we allow `IPv4 compatible` addresses as input. The rational is to be liberal in what we accept but more strict in what we generate (we only generate IPv4 mapped).

```
// We allow IPv4 Mapped (https://tools.ietf.org/html/rfc4291#section-2.5.5.1)
// and IPv4 compatible (https://tools.ietf.org/html/rfc4291#section-2.5.5.1).
// The IPv4 compatible is deprecated, but it allows parsing of plain IPv4 addressed into IPv6-Mapped addresses.
```